syntax = "proto3";

package animal_posts;

// Service definition
service AnimalPostService {
    rpc CreateAnimalPost (CreateAnimalRequest) returns (CreateAnimalResponse);
    rpc UpdateAnimalPost (UpdateAnimalRequest) returns (UpdateAnimalResponse);
    rpc GetAnimals (Empty) returns (AnimalListResponse);
    rpc DeleteAnimalPost (DeleteAnimalRequest) returns (DeleteAnimalResponse);
    rpc CheckStatus (Empty) returns (StatusResponse);
    rpc GetLoad(Empty) returns (LoadResponse);
}

// Request messages
message CreateAnimalRequest {
    string title = 1;
    string description = 2;
    string location = 3;
    string status = 4;
    string images = 5;
}

message CreateAnimalResponse {
    int32 postId = 1;
    string message = 2;
    int32 status_code = 3;
}

message UpdateAnimalRequest {
    int32 postId = 1;
    string title = 2;
    string description = 3;
    string location = 4;
    string status = 5;
    string images = 6;
}

message UpdateAnimalResponse {
    string message = 1;
    int32 status_code = 2;
}

message DeleteAnimalRequest {
    int32 postId = 1;
}

message DeleteAnimalResponse {
    string message = 1;
    int32 status_code = 2;
}

message Empty {}

message AnimalListResponse {
    repeated AnimalPost posts = 1;
    string source = 2;
    int32 status_code = 3;
}

message AnimalPost {
    int32 postId = 1;
    string title = 2;
    string description = 3;
    string location = 4;
    string status = 5;
    string images = 6;
}

message StatusResponse {
    string status = 1;
    int32 status_code = 2;
}

// The response message for GetLoad
message LoadResponse {
    int32 load = 1;
    int32 status_code = 2;
}
